client/a8c-for-agencies/sections/sites/sites-dashboard/index.tsx

```diff
@@ -181,6 +182,12 @@ export default function SitesDashboard() {
 		selectedText: selectedItem.label,
 	};
 
+	const isEmpty = ! isLoading && ! data?.sites?.length;
```

Comment:
Perhaps to make this work when a filter or search is active and also returning no results, we also need to check state to see if there is a filter or search active. `sitesViewState.search` and `sitesViewState.filters` can be checked to confirm that.
However, search and filter can be manually added to the URL, which means the filter and search values will be considered active even if an account has no sites initially. It would seem that doing a check and storing the number of total sites could be really useful here. Possibly for the sidebar too but I see that can only use the initial context. But the sidebar works correctly for me as it doesn't consider filters / search. This is what I see with a search of 'sss' entered:


<img width="1468" alt="Screenshot 2024-03-29 at 08 50 34" src="https://github.com/Automattic/wp-calypso/assets/16754605/ed3ae99e-8359-41bb-83fa-dd820f51bba7">

Reply:
Thanks for catching this. I've decided to make a `useNoActive` hook to provide the information for us to reliably check if the user has no sites. Let me know what your thoughts. https://github.com/Automattic/wp-calypso/pull/89034/commits/5755d4f7a66f1828ff1014d7d6592c82340ee598

Reply:
Great idea! The hook looks good to me too.